### ISSUE TITLE
Ontology Service Cache issue

### ISSUE THREAD

**Issue by kilchenmann** [dhlab-basel/Salsah#105]
After merging develop branch into my wip/fix-issue#99 branch i get the following error:

<img width="580" alt="screen shot 2017-11-17 at 14 46 03" src="https://user-images.githubusercontent.com/4253580/32950292-203cac0a-cba6-11e7-9df0-8aab6a2bd407.png">

@tobiasschweizer : Perhaps we have to sit together next week to find the bug...

**Comment by tobiasschweizer** [comment]
Ok!

**Comment by musicEnfanthen** [comment]
This type of error often occurs when you do not have imported the operator in question. So maybe it is enough to add
```
import 'rxjs/add/operator/flatMap'; 
``` 
(it is possible that the correct name of the operator is `mergeMap` of which `flatMap` is only an alias).

**Comment by tobiasschweizer** [comment]
It is indeed an alias :-)

Please see my comments in the source code: https://github.com/dhlab-basel/Salsah/blob/e2b461e145f6a17a5fcd17eec3132e6eeafb65c4/src/app/model/services/ontologycache.service.ts#L313

However, these are the import statements used for `ontologycache.service.ts` on  `develop`:

```
import {Injectable} from "@angular/core";
import {OntologyService} from "./ontology.service";
import {ApiServiceResult} from "./api-service-result";
import {Observable} from "rxjs/Observable";
import {AppConfig} from "../../app.config";
import {Utils} from "../../utils";
```

**Comment by kilchenmann** [comment]
@musicEnfanthen's solution works for me. I had to import the following rxjs methods:

`import 'rxjs/add/operator/mergeMap';`
`import 'rxjs/add/observable/fromPromise';`
`import 'rxjs/add/observable/of';`

**Comment by musicEnfanthen** [comment]
Good. The question is: Why did it work for @tobiasschweizer without these imports? Do you have an extra folder for all imported rxjs operators? (what is sometimes a recommended strategy). Or are there different versions of rxjs library? So far, I could not see any difference in `package.json` for this package in `develop` or on `wip/fix-issue#99`

**Comment by tobiasschweizer** [comment]
> Why did it work for @tobiasschweizer without these imports?

Excellent question! We would have to check if there are version changes etc. (package.json).
